pkg/verificationcache/metrics_reporter.go

```diff
@@ -0,0 +1,28 @@
+package verificationcache
+
+import "time"
+
+// MetricsReporter is an interface used by a verification cache to report various metrics related to its operation.
+// Implementations must be thread-safe.
+type MetricsReporter interface {
```

Comment:
**Question**: @rosecodym any reason we need this to be an interface? I'm seeing `InMemoryMetrics` as the only struct that implements this interface.

Reply:
Oh, good question. The plan is to use this seam to plug in a Prometheus-based reporter in enterprise. Rather than drape those metrics all over the OSS codebase I opted to create this interface, which looks silly on its own (as you've pointed out) but I hope will save some headaches in the long run.